static-analysis-plugin/plugin/src/test/groovy/com/novoda/staticanalysis/CheckstyleConfigurationTest.groovy

```diff
@@ -0,0 +1,158 @@
+package com.novoda.staticanalysis
+
+import com.novoda.test.Fixtures
+import com.novoda.test.TestProject
+import com.novoda.test.TestProjectRule
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+import static com.novoda.test.ResultSubject.assertThat
+
+@RunWith(Parameterized.class)
```

Comment:
Java and Android projects are different (from a setup point of view) just because they use a slightly different build script. Factoring out this via a test rule it's possible to run the very same suite against both type of setup.